mease_lab_to_nwb/convert_syntalos/syntalosrecordinginterface.py

```diff
@@ -30,38 +39,53 @@ def write_accelerometer_data(nwbfile: NWBFile, recording, stub_test: bool = Fals
         raise NotImplementedError("Unequal auxiliary channel unit types are not yet supported.")
     if not all_equal(accel_channel_sampling_rate):
         raise NotImplementedError("Unequal auxiliary channel sampling rates are not yet supported.")
-    if not isinstance(recording._recordings[0]._recording._raw_data[accel_channels[0]['name']], np.memmap):
-        raise NotImplementedError("Writing accelerometer data for non-memory maps is not supported!")
 
     conversion = channel_conversion[0]
+    accel_sampling_rate = accel_channel_sampling_rate[0]
     all_memmaps = [
         [x._recording._raw_data[accel_channels[j]['name']].flatten() for j in range(accel_channels.size)]
-        for x in recording._recordings
+        for x in this_recording._recordings
     ]
 
     all_accel_data = np.concatenate(np.moveaxis(np.array(all_memmaps), 2, 1))
     if stub_test:
         all_accel_data = all_accel_data[:100, :]
 
-    nwbfile.add_acquisition(
-        TimeSeries(
-            name="Accelerometer",
-            description="Data recorded from auxiliary channels from an intan device, tracking acceleration.",
-            data=H5DataIO(all_accel_data, compression="gzip"),
-            rate=accel_channel_sampling_rate[0],  # TODO: replace this using tsync mapping?
-            unit=accel_channel_units[0],
-            resolution=np.nan,
-            conversion=conversion
-        )
+    tseries_kwargs = dict(
+        name="Accelerometer",
+        description="Data recorded from auxiliary channels from an intan device, tracking acceleration.",
+        data=H5DataIO(all_accel_data, compression="gzip"),
+        unit=accel_channel_units[0],
+        resolution=np.nan,
+        conversion=conversion
     )
+    if not use_timestamps:
+        tseries_kwargs.update(rate=accel_sampling_rate)
+    else:
+        accel_timestamps = recording.frame_to_time(
+            np.arange(
+                0,
+                this_recording.get_num_frames(),
+                this_recording.get_sampling_frequency() / accel_sampling_rate
+            ).astype(int)
+        )
+        tseries_kwargs.update(timestamps=H5DataIO(accel_timestamps, compression="gzip"))
+
+    nwbfile.add_acquisition(TimeSeries(**tseries_kwargs))
 
 
 class SyntalosRecordingInterface(BaseRecordingExtractorInterface):
     """Conversion class for Syntalos Recording + Accelerometer."""
 
     RX = SyntalosRecordingExtractor
 
-    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False, add_accelerometer: bool = True):
+    def get_metadata(self):
+        intan_filepath = [x for x in Path(self.source_data['folder_path']).iterdir() if x.suffix == ".rhd"][0]
+        temp_intan_interface = IntanRecordingInterface(file_path=intan_filepath)
+        return temp_intan_interface.get_metadata()
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False, add_accelerometer: bool = True,
```

Comment:
metadata is not named properly in docstring

Reply:
Fixed